tests/Go/Functional/BaseFunctionalTest.php

```diff
@@ -158,10 +158,87 @@ protected function assertClassIsWoven($class, $message = '')
+    {
+        self::assertThat(new ClassAdvisorIdentifier($class, $staticMethodName, $advisorIdentifier, AspectContainer::STATIC_METHOD_PREFIX, $index), new ClassMemberWovenConstraint($this->configuration), $message);
+    }
+
+    /**
+     * Assert that class static method is not woven.
+     *
+     * @param string $class Full qualified class name which is subject of weaving.
+     * @param string $staticMethodName Name of static method.
+     * @param string $advisorIdentifier Expected advisor identifier to be registered within proxy class.
+     * @param string $message Assertion info message.
+     */
+    protected function assertStaticMethodNotWoven($class, $staticMethodName, $advisorIdentifier, $message = '')
+    {
+        self::assertThat(new ClassAdvisorIdentifier($class, $staticMethodName, $advisorIdentifier, AspectContainer::STATIC_METHOD_PREFIX), new ClassMemberNotWovenConstraint($this->configuration), $message);
+    }
+
+    /**
+     * Assert that class method is woven.
+     *
+     * @param string $class Full qualified class name which is subject of weaving.
+     * @param string $methodName Name of method.
+     * @param string $advisorIdentifier Expected advisor identifier to be registered within proxy class.
+     * @param null|int $index Index of advisor identifier, or null if order is not important.
+     * @param string $message Assertion info message.
+     */
+    protected function assertMethodWoven($class, $methodName, $advisorIdentifier, $index = null, $message = '')
+    {
+        self::assertThat(new ClassAdvisorIdentifier($class, $methodName, $advisorIdentifier, AspectContainer::METHOD_PREFIX, $index), new ClassMemberWovenConstraint($this->configuration), $message);
+    }
+
+    /**
+     * Assert that class method is not woven.
+     *
+     * @param string $class Full qualified class name which is subject of weaving.
+     * @param string $methodName Name of method.
+     * @param string $advisorIdentifier Expected advisor identifier to be registered within proxy class.
+     * @param string $message Assertion info message.
+     */
+    protected function assertMethodNotWoven($class, $methodName, $advisorIdentifier, $message = '')
+    {
+        self::assertThat(new ClassAdvisorIdentifier($class, $methodName, $advisorIdentifier, AspectContainer::METHOD_PREFIX), new ClassMemberNotWovenConstraint($this->configuration), $message);
+    }
+
+    /**
+     * Assert that class property is woven.
+     *
+     * @param string $class Full qualified class name which is subject of weaving.
+     * @param string $propertyName Property name.
+     * @param string $advisorIdentifier Expected advisor identifier to be registered within proxy class.
+     * @param null|int $index Index of advisor identifier, or null if order is not important.
+     * @param string $message Assertion info message.
+     */
+    protected function assertPropertyWoven($class, $propertyName, $advisorIdentifier, $index = null, $message = '')
```

Comment:
Just quickly think about making advosorIdentifier optional. Because we can just check that there is at least key for our property. Same can be applied to methods as well. What do you think?